### ISSUE TITLE
removing specification requirement of .desktop files

### ISSUE THREAD

**Issue by CosmicToast** [TheAssassin/type3-runtime#4]
`.desktop` files only make sense for *interactive* (readline-like, tui and gui) applications.
This means that non-interactive applications can't be easily packaged in an appimage, which I think is a shame.
For example, consider the go runtime: even with Terminal=true, it doesn't make much sense.
There's no icon for go(1) or gofmt(1).
There's no point in launching them from a menu.

However, you cannot distribute them as standalone executables, because they fundamentally depend on GOROOT(_FINAL) to function, since that's where the stdlib etc is kept, even though they have no external dependencies besides, well, themselves and GOROOT.

So to package a golang runtime, you have to procure an unofficial application icon from somewhere and make a nonsense `.desktop` file just to comply to the specification.
Wouldn't it instead, perhaps, be preferable to make .desktop files optional, and simply not perform any integration work if there isn't one specified?

**Comment by TheAssassin** [comment]
I tend to agree. For many applications, e.g., linuxdeploy, appimagetool, all linuxdeploy plugins, appimagecraft etc. a desktop file is not needed at all. Those are also the kinds of applications you would *not* put into a software store or anything, where the desktop file information would be processed.

**Comment by probonopd** [comment]
Within the AppImage ecosystem, we are using desktop files for various tasks, including determining the name of the AppImage, storing a description string of what the application/CLI tool does, and for displaying metadata in App Centers. I don't see why it is a problem to create a desktop file for CLI tools.
Also, not having an icon is not something every AppImage-using tool should need to be able to handle, it only makes the code more complex. By the way, Go does have an icon and this is how my Go AppImage looks:

![](https://user-images.githubusercontent.com/2480569/79489243-848bbc80-800a-11ea-9fb2-29f82ef986ff.png)

As per the AppImage specification, an AppImage

> MUST contain a .DirIcon file as per the AppDir specification which SHOULD be a 256x256 PNG file.

So this is even coming from the AppDir specification which we have inherited from ROX.

> Those are also the kinds of applications you would not put into a software store or anything, where the desktop file information would be processed.

Why would you _not_ put those into App Centers?

**Comment by CosmicToast** [comment]
> I don't see why it is a problem to create a desktop file for CLI tools.

Because you end up having to fill something up that is of no practical to use (except to the appimage ecosystem itself). It's basically being created purely to please the runtime.

> As per the AppImage specification

Certainly. The discussion here is about potentially changing the specification, since we're at a point where there is an opportunity to do that.

> Within the AppImage ecosystem, we are using desktop files for various tasks, including determining the name of the AppImage, storing a description string of what the application/CLI tool does, and for displaying metadata in App Centers.

I think appimage-specific metadata should live in the appimage header, rather than be tacked-on to a different specification that is intended for something else.

Worse case scenario I'll just go my own way without having compatibility with appimages (besides the basic user UX), but I do think there's value in cooperating instead.
